packages/debugger/lib/data/sagas/index.js

```diff
@@ -1074,19 +1074,27 @@ export function* decode(
     let response;
     switch (request.type) {
       case "storage":
-        //the debugger supplies all storage it knows at the beginning.
-        //any storage it does not know is presumed to be zero.
-        //(unlesss indicateUnknown is passed, in which case we use
-        //null as a deliberately invalid response)
-        response = indicateUnknown ? null : ZERO_WORD;
+        if (storageLookup) {
+          response = yield* evm.requestStorage(request.slot);
+        } else if (indicateUnknown) {
+          //the debugger supplies all storage it knows at the beginning.
+          //so anything it gets a request for can be presumed to be
+          //unknown.
+          response = null;
+        } else {
+          //if indicateUnknown isn't set, all unknown storage is assumed
+          //to be zero.
+          response = ZERO_WORD;
+        }
         break;
       case "code":
-        response = yield* requestCode(request.address);
+        response = yield* evm.requestCode(request.address);
```

Comment:
like, of course you ask the evm to request storage or code!

Reply:
:)